src/main/java/duke/parsers/InputParser.java

```diff
@@ -0,0 +1,205 @@
+     * @throws DukeException
+     */
+    public static void inputChecker(String[] str, String task) throws DukeException {
+
+        if (str.length < 2) {
+            throw new DukeException("☹ OOPS!!! The description of a " + task + " cannot be empty.");
+        }
+    }
+
+    /***
+     * receives string of date and time, returns Date object
+     * @param str
+     * @return Date
+     * @throws DukeException
+     */
+    public static Date dateParser(String str) throws DukeException {
+        if (str.length() < 15) {
+            throw new DukeException("Invalid date, must be of the form dd/mm/yyyy hhmm");
+        }
+        String newStr = str.substring(0, 13) + ":" + str.substring(13);
+        System.out.println(newStr);
+        DateFormat df = new SimpleDateFormat("dd/MM/yyyy HH:mm");
+        try {
+            Date d1 = df.parse(newStr);
+            return d1;
+        } catch (Exception e) {
+            System.out.println(e);
+            throw new DukeException("Invalid date");
+        }
+    }
+
+    /***
+     * parses user input
+     * @param input
+     * @param toStore
+     * @return truw if user has not exited, false if user has exited chatbot
+     */
+
+    public static boolean parse(String input, boolean toStore) {
+        String[] splitStr = input.trim().split("\\s+");
+
+        if (input.equals("bye")) {
+            return false;
+        }
+        //list out each task from Duke.tasks ArrayList
+        else if (input.equals("list")) {
+            System.out.println("  Here are the tasks in your list:");
+            for (int i = 0; i < Task.getSize(); i++) {
+                int index = i + 1;
+                System.out.println("  " + index + "." + tasks.get(i).toString());
+            }
+        }
+        //create Todo object
+        else if (splitStr[0].equals("todo")) {
+            try {
+                inputChecker(splitStr, "todo");
+                Todo t = new Todo(input.substring(5));
+                tasks.add(t);
+                if (toStore){
+                    Storage.saveTasks(tasks);
+                };
+            } catch (DukeException e) {
+                System.out.println(e.getMessage());
+            }
+        }
+        //create deadline object, splitting the due date by "/" and stripping off the by:
+        else if (splitStr[0].equals("deadline")) {
+            try {
+                inputChecker(splitStr, "deadline");
+                String[] deadlineArr = input.split("/by ");
+                Date deadline = dateParser(deadlineArr[1]);
+
+                Deadline d = new Deadline(deadlineArr[0].substring(9), deadline);
+                tasks.add(d);
+                if (toStore){
+                    Storage.saveTasks(tasks);
+                };
+            } catch (DukeException e) {
+                System.out.println(e.getMessage());
+            }
+        }
+        //create event object, splitting the due date by "/" and stripping off the to: and from:
```

Comment:
nice comment explaining each if block parts accordingly, code also seems to be in a logical order, easy for me to understand, and also following the KISS (Keep It Simple, Stupid) convention.